core/trino-spi/src/main/java/io/trino/spi/block/BlockBuilder.java

```diff
@@ -96,4 +96,14 @@ default Block getPositions(int[] visiblePositions, int offset, int length)
      * Creates a new block builder of the same type based on the current usage statistics of this block builder.
      */
     BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus);
+
+    /**
+     * Returns a block that has an appended null at the end, no matter if the original block has null or not.
+     * The original block won't be modified.
+     */
+    @Override
+    default Block newBlockWithAppendedNull()
```

Comment:
Actually - should it just be called `buildWithAppendedNull`?

Reply:
Also if you could calrify in the javadoc that it may not be not supperted and when is that the case.

Reply:
> Actually - should it just be called buildWithAppendedNull

hmm, since we're adding this method on `Block` interface, I feel `build*` will be misleading, as a Block implementation can be already built. I like `copyWithAppendedNull` too, updating.